### PR TITLE
update snapshot version to v2306

### PR DESCRIPTION
**Pull request by nvliyuan**
update cuML/udf/cuSpatial pom/scripts version to v2306-snapshot
Signed-off-by: liyuan <yuali@nvidia.com>

### CODE REVIEW

File: examples/ML+DL-Examples/Spark-cuML/pca/pom.xml [hunk 1]
```diff
@@ -21,7 +21,7 @@
     <groupId>com.nvidia</groupId>
     <artifactId>PCAExample</artifactId>
     <packaging>jar</packaging>
-    <version>23.04.0-SNAPSHOT</version>
+    <version>23.06.0-SNAPSHOT</version>
```
Comment (pxLi) [inline examples/ML+DL-Examples/Spark-cuML/pca/pom.xml:24]: PCA was not updated recently. ML repo has been refactored to only python pkg for 23.04

File: examples/ML+DL-Examples/Spark-cuML/pca/pom.xml [hunk 2]
```diff
@@ -51,7 +51,7 @@
         <dependency>
             <groupId>com.nvidia</groupId>
             <artifactId>rapids-4-spark-ml_2.12</artifactId>
-            <version>23.04.0-SNAPSHOT</version>
+            <version>23.06.0-SNAPSHOT</version>
```
Comment (pxLi) [inline examples/ML+DL-Examples/Spark-cuML/pca/pom.xml:54]: PCA was not updated recently. ML repo has been refactored to only python pkg for 23.04

Comment (nvliyuan) [reply examples/ML+DL-Examples/Spark-cuML/pca/pom.xml]: thanks for the info, updated

**Comment by NvTimLiu** [comment]
Please help to update udf files: 


https://github.com/NVIDIA/spark-rapids-examples/blob/branch-23.06/examples/UDF-Examples/RAPIDS-accelerated-UDFs/pom.xml#L28

https://github.com/NVIDIA/spark-rapids-examples/blob/branch-23.06/examples/UDF-Examples/RAPIDS-accelerated-UDFs/pom.xml#L40

23.04 in https://github.com/NVIDIA/spark-rapids-examples/blob/branch-23.06/examples/UDF-Examples/RAPIDS-accelerated-UDFs/src/main/cpp/CMakeLists.txt

**Comment by nvliyuan** [comment]
> Please help to update udf files:

updated, thanks

**Review by NvTimLiu — APPROVED** [review]

**Pull request merged**